index.js

```diff
@@ -76,7 +76,15 @@ module.exports = fp(function from (fastify, opts, next) {
         // detect if body should be encoded as JSON
         // supporting extended content-type header formats:
         // - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type
-        const shouldEncodeJSON = contentType.toLowerCase().indexOf('application/json') === 0
+        const contentTypesToEncode = new Set([
+          'application/json',
+          ...(opts.contentTypesToEncode || [])
+        ])
+        const lowerCaseContentType = contentType.toLowerCase()
+        const plainContentType = lowerCaseContentType.indexOf(';') > -1
+          ? lowerCaseContentType.slice(0, lowerCaseContentType.indexOf(';'))
+          : lowerCaseContentType
+        const shouldEncodeJSON = contentTypesToEncode.has(plainContentType)
```

Comment:
I meant that this needs to go on like 19 and it should be an option of the plugin.

Reply:
The code that you selected above cannot be moved up as it depends on the current request for the content type. I think  you meant for me to move the `contentTypesToEncode` option onto the register options rather than the `decorateReply`, which I have now done.

Reply:
This does what I meant.